aegis-web/yo/app/views/datasetsBrowser.html

```diff
@@ -52,18 +51,61 @@
+                      </i>
+                      <i ng-if="!mainCtrl.unreadMessages" class="fa fa-envelope fa-lg" title="No unread message"></i>
+                      &nbsp;Messages
+                    </a>
+                    <ul class="dropdown-menu dropdown-messages" >
+                      <li class="dropdown-header text-muted" ng-if="mainCtrl.messages.length"><h4>Latest messages</h4></li>
+                      <li class="dropdown-header text-muted" ng-if="!mainCtrl.messages.length"><h4>Inbox empty.</h4></li>
+                      <li ng-repeat="msg in mainCtrl.messages| limitTo:5">
+                        <div>
+                          <div ng-click="mainCtrl.openMessageModal(msg)">                    
+                            <div class="message-subject text-muted">
+                              <i ng-if="msg.unread" class="fa fa-circle pull-left" style="color:#f89406; margin-top: 4px;" title="Unread message"></i>
+                              <i ng-if="!msg.unread" class="fa fa-circle pull-left" style="color:grey; margin-top: 4px;" title="Read message"></i>
+                              {{msg.subject}}</div>
+                            <div class="text-muted">{{msg.dateSent| date:'medium'}}</div>
+                          </div>
+                          <div ng-click="mainCtrl.openMessageModal(msg)" class="message-preview">{{msg.preview| limitTo:80}} {{msg.preview.length > 80 ? ' ...' : ''}}</div>
+                          <a ng-if="msg.path" href="#!/{{msg.path}}"> go to requested path.</a>
+                        </div>
+                      </li>
+                      <a class="pull-right" ng-if="mainCtrl.messages.length" ng-click="mainCtrl.openMessageModal()" style="text-decoration:none;"> See all messages</a>
+                      <a class="pull-right" ng-if="!mainCtrl.messages.length" ng-click="mainCtrl.openMessageModal()" style="text-decoration:none;"> Goto messages</a>
+                    </ul>
+                  </li>          
+                </ul>
+              </div>
+            </div>
+          <div class="row content-utility-bar content-utility-bar-actions" ng-if="!mainCtrl.showSearchPage">
               <div class="col-md-3">
                 <h4 style="margin-top: 10px; margin-bottom: 5px"><a ng-click="datasetsCtrl.goToDataSetsDir()">DataSets</a>
                 <span ng-show="datasetsCtrl.pathArray.length > 10">...</span>
                 <span data-ng-repeat="path in datasetsCtrl.pathArray.slice(datasetsCtrl.pathArray.length - 1 - datasetsCtrl.breadcrumbLen()) track by $index">
-                  <span> > </span><a ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</a>
+                  <span> > </span><span ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</span>
```

Comment:
Also, i just noticed this change. You shouldn't camouflage clickable links to text. Keep the a tag.

Reply:
Are you changing this?

Reply:
i kept <a ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</a>

Reply:
I was talking about the change from `a` to `span`. You should keep the `a` tag instead

Reply:
sorry I kept <a ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</a> like before

Reply:
![image](https://user-images.githubusercontent.com/1640100/56659503-9b206180-669d-11e9-9a09-c070adbb5c8f.png)
Revert this change

Reply:
I did, but you don't see it yet

Reply:
commit and push so that i can merge this PR

Reply:
![Schermata 2019-04-24 alle 14 33 35](https://user-images.githubusercontent.com/47423916/56659643-f7838100-669d-11e9-8953-d84fe941e363.png)

I don't know why it doesn't show the right message

Reply:
you need to wrap your text with backtick ( ` ) when whitting code in github comments. If it's a code block, you need triple backtick ( ``` )
Read more:

- https://daringfireball.net/projects/markdown/syntax#code
- https://help.github.com/en/articles/creating-and-highlighting-code-blocks

Reply:
thanks